0-module-and-module-source.emu

```diff
@@ -548,8 +550,11 @@ location: https://tc39.es/proposal-compartments/
                 1. Else, if Type(_importMeta_) is not Object, then
                   1. Throw a *TypeError* exception.
                 1. Perform ? RequireInternalSlot(_moduleSource_, [[ModuleSource]]).
+                1. Let _evalRealm_ be the current Realm Record.
+                1. NOTE: Module Instances created from this constructor must use a valid source.
+                1. Perform ? HostEnsureCanCompileStrings(_evalRealm_).
```

Comment:
@nicolo-ribaudo this is important so we can serialize and share a ModuleSource instance with another realm, and evaluate it there if possible without requiring string evaluation privileges on the realm creating the source.

Reply:
Also I believe the Host operation has the wrong (or misleading) name since the problem is not really compilation IMO, but evaluation. We should get some clarifications from implementers.

Reply:
cc @Jack-Works

Reply:
This is somehow like you create a new Function via eval and call it in a CSP-enabled Realm. It's OK. You should put the check in the ModuleSource constructor.

Reply:
> Creating a ModuleSource is not problematic

I just realized that this _is_ problematic: for layer 1, hosts must parse the string passed to `ModuleSource` to provide the static analysis info, and thus it requires them to ship a complete parser.